### PR TITLE
build: update to @angular/cli@11.0.0-rc.1

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

File: integration/dynamic-compiler/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "main": "index.js",
   "scripts": {
-    "build": "npm run clean && npm run ngc && npm run rollup && npm run rollup:lazy && npm run es5 && npm run es5:lazy",
+    "build": "yarn clean && yarn ngc && yarn rollup && yarn rollup:lazy && yarn es5 && yarn es5:lazy",
```
Comment (IgorMinar) [inline integration/dynamic-compiler/package.json:6]: this is just a drive-by cleanup. for some reason without these changes I could not run these tests locally on linux likely due to bazel sandboxing.

**Review by IgorMinar — COMMENTED** [review]
It appears that we have a regression in the CLI, somewhere between CLI v10-rc.2 and v11-rc.0. More details in the individual diff comments.

**Review by petebacondarwin — DISMISSED** [review]
The `//packages/core/schematics/test:test` job seems to be a legit failure...

File: package.json [hunk 2]
```diff
@@ -156,7 +156,7 @@
   },
   "// 2": "devDependencies are not used under Bazel. Many can be removed after test.sh is deleted.",
   "devDependencies": {
-    "@angular/cli": "9.1.0",
+    "@angular/cli": "11.0.0-rc.0",
```
Comment (alan-agius4) [inline package.json:159]: Can you kindly also bump `@schematics/angular`

File: goldens/size-tracking/integration-payloads.json [hunk 3]
```diff
@@ -39,19 +39,19 @@
...
     }
   },
   "cli-hello-world-lazy-rollup": {
     "master": {
       "uncompressed": {
         "runtime-es2015": 2289,
-        "main-es2015": 217879,
-        "polyfills-es2015": 36723,
-        "5-es2015": 781
+        "main-es2015": 222611,
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply goldens/size-tracking/integration-payloads.json:52]: I looked at the extra parenthesis issue and it looks to be a terser regression, Will create a reproduction and open an issue.

Regarding the extra RxJs code, it does look like @clydin is on it as he did create a PR https://github.com/angular/angular-cli/pull/19221

[the remark this answers is not in the document]
Comment (alan-agius4) [reply goldens/size-tracking/integration-payloads.json:52]: So, it looks like actually the parenthesis change was intended as it was a previously a bug that these were not added for lambas since `wrap_func_args` was enabled by default.

https://github.com/terser/terser/commit/66c3a5ce66c7873bded8c6466e72027f9816c548

I'll do a change in the CLI to turn of `wrap_func_args`.

[the remark this answers is not in the document]
Comment (clydin) [reply goldens/size-tracking/integration-payloads.json:52]: Based on the following article, the “wrap_func_args” should probably be disabled either way.  The option makes files slightly larger in the attempt to improve runtime performance by trying to trigger special cases in the parser.  However,  this apparently is not recommended in newer versions of V8 and can lead to worse performance in some situations.

Article: https://v8.dev/blog/preparser#pife

[the remark this answers is not in the document]
Comment (alan-agius4) [reply goldens/size-tracking/integration-payloads.json:52]: Thanks for sharing that @clydin! Really interesting

File: goldens/size-tracking/integration-payloads.json [hunk 4]
```diff
@@ -12,8 +12,8 @@
     "master": {
       "uncompressed": {
         "runtime-es2015": 1485,
-        "main-es2015": 17049,
-        "polyfills-es2015": 36657
+        "main-es2015": 17308,
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply goldens/size-tracking/integration-payloads.json:15]: Should be fixed by @clydin change.

File: goldens/size-tracking/integration-payloads.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (IgorMinar) [reply goldens/size-tracking/integration-payloads.json:52]: Thanks @clydin!

File: package.json (hunk 2 above)
Comment (IgorMinar) [reply package.json:159]: done!

**Review by kyliau — APPROVED** [review]

**Comment by kyliau** [comment]
@angular/cli@11.0.0-rc.1 is released, it should fix the size regression

**Review by alxhub — APPROVED** [review]

**Review by clydin — APPROVED** [review]

**Review by alan-agius4 — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: size-tracking

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by jelbourn — APPROVED** [review]
LGTM

Reviewed-for: size-tracking

**Review by alxhub — APPROVED** [review]
